packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -187,12 +188,75 @@ private Query getQuery(Map<String, Object> arguments) {
       query = query.orderBy(orderByFieldName, direction);
     }
     @SuppressWarnings("unchecked")
+    ArrayList startAtDocument = (ArrayList) parameters.get("startAtDocument");
```

Comment:
Feels like the below 4 blocks of code of 4 different querying are pretty identical, can refactor them out to a private method.

Reply:
Good point. I have done this, PTAL